python/src/aiconfig/editor/server/server_utils.py

```diff
@@ -77,6 +79,24 @@ def convert_to_mode(
         return value
 
 
+class StartServerConfig(core_utils.Record):
+    server_port: int = 8080
+    log_level: str | int = "INFO"
+    server_mode: ServerMode = ServerMode.PROD
+    parsers_module_path: str = "aiconfig_model_registry.py"
+
+    @field_validator("server_mode", mode="before")
+    def convert_to_mode(
+        cls, value: Any
+    ) -> ServerMode:  # pylint: disable=no-self-argument
+        if isinstance(value, str):
+            try:
+                return ServerMode[value.upper()]
+            except KeyError as e:
+                raise ValueError(f"Unexpected value for mode: {value}") from e
+        return value
```

Comment:
nit: we can dedup with EditServerConfig either by making aiconfig_path optional or nesting the common fields in a shared sub-record type. Also I think we have enough of these enum field validators now to justify a util function.

Reply:
https://github.com/lastmile-ai/lastmile-utils/issues/31

Reply:
I thought about that, but figured there may be other things to add to the `start` command in the future that the edit command doesn't need, so opted to keep them separate